src/routes/rosters/+page.svelte

```diff
@@ -1,63 +1,97 @@
+      newTeamName = "";
+      newTeamHometown = "";
+      newCoachName = "";
+  
+      // Close the modal
+      closeAddModal();
+    }
+  </script>
+  
+  <title>Rosters</title>
+  
+  <div>
+    <section class="rosters-dash">
+      <h1>Rosters</h1>
+      <h3>A List of All Teams on the Website</h3>
     </section>
-
+  
     <section class="list-section-1">
-        <div class = "buttons">
-            <button on:click={openAddModal} type="button">Add Teams</button>
-        </div>
-        <div>
-            <h1>All Teams</h1>
-        </div>
-        <div>
-            {#each teams as team }
-            <a href='/rosters/{team}'>
-                <div class = "game">
-                    <h3>{team}</h3>
-                </div>
-            </a>
-            {/each}
-        </div>
-
-    </section>
-</div>
-
- {#if showAddModal}
-        <div class="modal-backdrop">
-            <!-- Modal Content -->
-            <div class="modal-content">
-                <h2>Sign In</h2>
-                <form>
-                    <div class="form-group">
-                        <label for="team-name">Team Name:</label>
-                        <input type="text" id="team-name" required>
-                    </div>
-                    <div class="form-group">
-                        <label for="hometown">Hometown:</label>
-                        <input type="text" id="hometown" required>
-                    </div>
-                    <div class="form-group">
-                        <label for="coach-name">Coach:</label>
-                        <input type="text" id="coach-name" required>
-                    </div>
-                    <div class="modal-actions">
-                        <button type="button" on:click={closeAddModal} class="cancel-button">Cancel</button>
-                        <button type="submit" class="sign-in-button">Add</button>
-                    </div>
-                </form>
+      <div class="buttons">
+        <button on:click={openAddModal} type="button">Add Teams</button>
+      </div>
+      <div>
+        <h1>All Teams</h1>
+      </div>
+      <div>
+        {#each teams as team}
+          <!-- Using team.name in the href and display -->
+          <a href="/rosters/{team.name}">
+            <div class="game">
+              <h3>{team.name}</h3>
+              <p>{team.hometown}</p>
+              <p>Coach: {team.coach}</p>
             </div>
-        </div>
-    {/if}
\ No newline at end of file
+          </a>
+        {/each}
+      </div>
+    </section>
+  </div>
+  
+  {#if showAddModal}
+    <div class="modal-backdrop">
+      <div class="modal-content">
+        <h2>Add Team</h2>
+        <!-- Bind input values and handle form submission -->
+        <form on:submit|preventDefault={handleAddTeam}>
+          <div class="form-group">
+            <label for="team-name">Team Name:</label>
+            <input type="text" id="team-name" bind:value={newTeamName} required>
```

Comment:
The way forms classify their field values is by using an HTML tag called `name`, so instead of using id (which is useless here because you never access any of these elements individually by id), you should do this:
```suggestion
            <input type="text" name="team-name" bind:value={newTeamName} required>
```